corehq/apps/reports/models.py

```diff
@@ -561,3 +562,19 @@ def delete_user(self, id):
             {}
         )
         return True
+
+
+class QueryStringHash(models.Model):
+    date_created = models.DateTimeField(auto_now_add=True)
+    last_accessed = models.DateTimeField(auto_now=True)
+    query_hash = models.CharField(max_length=64, db_index=True)
+    query_string = models.TextField()
+
+    @staticmethod
+    def create_hash_from_string(query_string):
+        return hashlib.sha256(query_string.encode()).hexdigest()
```

Comment:
Would it work to return a UUID instead of hashing the query string? I think that would reduce the chances of a collision, however unlikely that may seem.

Reply:
Changed to generate UUIDs! also changed variable names to reflect this (replaced 'hash' with 'id's)